backend/amd64/proc.ml

```diff
@@ -133,10 +133,22 @@ let num_available_registers = [| 13; 16 |]
 
 let first_available_register = [| 0; 100 |]
 
-let register_name r =
-  if r < 100 then int_reg_name.(r)
-  else if r < 200 then float_reg_name.(r - 100)
-  else Misc.fatal_errorf "Register of unknown class (%d)" r
+let register_name ty r =
+  (* If the ID doesn't match the type, the array access will raise. *)
+  match ty with
+  | Int | Addr | Val ->
+    int_reg_name.(r - first_available_register.(0))
+  | Float ->
+    float_reg_name.(r - first_available_register.(1))
+  | Vec128 ->
+    if not !simd_regalloc_support then
+    Misc.fatal_error "SIMD register allocation is not enabled.";
+    float_reg_name.(r - first_available_register.(1))
+
+let register_name_lossy idx =
+  if idx < 100 then int_reg_name.(idx)
+  else if idx < 200 then float_reg_name.(idx - 100)
+  else Misc.fatal_errorf "Unknown register ID %d" idx
```

Comment:
We should replace the magic 100 and 200 constants with `typ` as in register_name, and we should keep this "lossy" business inside the validator. Perhaps the right thing to do is to record `typ` in `Description.Register.t`.

Reply:
Refactored printing in validator to not require this